packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -374,18 +375,55 @@ export default class WalletService {
       throw new UsedName('Wallet')
     }
 
-    const wallet = this.fromJSON({ ...props, id: uuid() })
+    const id = uuid()
+
+    const wallet = this.fromJSON({ ...props, id })
 
     if (!wallet.isHardware()) {
       wallet.saveKeystore(props.keystore!)
     }
 
+    if (this.getAll().find(item => item.extendedKey === props.extendedKey)) {
+      this.importedWallet = wallet
+      throw new ImportingExitingWallet(JSON.stringify({ extendedKey: props.extendedKey, id }))
+    }
+
     this.listStore.writeSync(this.walletsKey, [...this.getAll(), wallet.toJSON()])
 
     this.setCurrent(wallet.id)
     return wallet
   }
 
+  public replace = async (existingWalletId: string, importedWalletId: string) => {
+    const wallet = this.get(existingWalletId)
+    if (!wallet || !this.importedWallet) {
+      throw new WalletNotFound(existingWalletId)
+    }
+
+    const newWallet = this.importedWallet?.toJSON()
+    if (importedWalletId !== newWallet.id) {
+      throw new WalletNotFound(importedWalletId)
+    }
+    if (wallet.toJSON().extendedKey !== newWallet.extendedKey) {
```

Comment:
It is recommended to use `bytes.equals(wallet.extendedKey, newWallet.extendedKey)` to avoid the case issue, like `"0xFF" !== "0xff"`

Reply:
These two values should be exactly the same, without ignoring case differences.